Api/Validator.php

```diff
@@ -0,0 +1,153 @@
+<?php
+    
+    namespace Metrilo\Analytics\Api;
+    
+    class Validator {
+        private $logDestination = __DIR__ . '/MetriloApiErrors.log'; // construct log path in ajax controller and pass it as constructor param for Validator class
+        
+        public function logger($error, $logDestination) {
+            return error_log($error, 3, $logDestination);
+        }
+        
+        public function validateCustomer($customer) {
```

Comment:
Everything here seems very complicated. Have a look at how this library handles validation https://github.com/davidecesarano/Validation

Reply:
Done, code has been refactored.